swarm_dial.go

```diff
@@ -307,10 +313,18 @@ func (s *Swarm) dial(ctx context.Context, p peer.ID) (*Conn, error) {
 	/////////
 
 	// try to get a connection to any addr
-	connC, err := s.dialAddrs(ctx, p, goodAddrsChan)
-	if err != nil {
-		logdial["error"] = err.Error()
-		return nil, err
+	connC, dialErr := s.dialAddrs(ctx, p, goodAddrsChan)
+	if dialErr != nil {
+		logdial["error"] = dialErr.Cause.Error()
+		if dialErr.Cause == context.Canceled {
```

Comment:
This fixes a bug introduced by the multierror patch.

Reply:
Nice catch.